content/docs/rfcs/17/README.md

```diff
@@ -94,6 +94,9 @@ Proof generation relies on the knowledge of Merkle tree root `merkle_root` and `
 Getting access to the Merkle tree can be done in various ways. 
 One way is that all the peers construct the tree locally.
 This can be done by listening to the registration and deletion events emitted by the membership contract.
+Peers MUST update the local Merkle tree on a per-block basis.
```

Comment:
If we use the term `MUST` for per block basis processing, I'd suggest making sure we use `MUST` for the size of the window of acceptable roots for consistency and interoperability sake.

Reply:
Both use cases are different in my opinion. `MUST` is used for per block basis processing since that would lead to invalid proofs being generated, and valid proofs invalidated on a peer. 
It should be upto the peer to decide how many roots they would like to validate, because in this case, invalid proofs (if they are too old for other peers) may be validated, at the cost of the peers resources. wdyt?